src/Cell.tsx

```diff
@@ -1,58 +1,59 @@
-// @flow
-
-import React, { PureComponent } from "react";
+import React, { PureComponent, SyntheticEvent } from "react";
 import classnames from "classnames";
 import { connect } from "unistore/react";
-import type { Parser as FormulaParser } from "hot-formula-parser";
+
 import * as PointSet from "./point-set";
 import * as PointMap from "./point-map";
 import * as Matrix from "./matrix";
 import * as Types from "./types";
 import * as Actions from "./actions";
-import { isActive, getOffsetRect } from "./util";
-
-type StaticProps<Data, Value> = {|
-  row: number,
-  column: number,
-  DataViewer: Types.DataViewer<Data, Value>,
-  getValue: Types.getValue<Data, Value>,
-  formulaParser: FormulaParser
-|};
-
-export type { StaticProps as Props };
-
-type State<Data> = {|
-  selected: boolean,
-  active: boolean,
-  copied: boolean,
-  dragging: boolean,
-  mode: Types.Mode,
-  data: ?Data,
-  _bindingChanged: ?Object
-|};
-
-type Handlers = {|
-  select: (cellPointer: Types.Point) => void,
-  activate: (cellPointer: Types.Point) => void,
-  setCellDimensions: (point: Types.Point, dimensions: Types.Dimensions) => void
-|};
-
-type Props<Data, Value> = {|
-  ...StaticProps<Data, Value>,
-  ...State<Data>,
-  ...Handlers
-|};
-
-export class Cell<Data: Types.CellBase, Value> extends PureComponent<
+import { getOffsetRect, isActive } from "./util";
+
+export type StaticProps<Data, Value> = {
+  row: number;
+  column: number;
+  DataViewer: Types.DataViewer<Data, Value>;
+  getValue: Types.getValue<Data, Value>;
+  formulaParser: any;
+};
+
+type State<Data> = {
+  selected: boolean;
+  active: boolean;
+  copied: boolean;
+  dragging: boolean;
+  mode: Types.Mode;
+  data?: Data;
+  _bindingChanged?: Object;
+};
+
+type Handlers = {
+  select: (cellPointer: Types.IPoint) => void;
+  activate: (cellPointer: Types.IPoint) => void;
+  setCellDimensions: (
+    point: Types.IPoint,
+    dimensions: Types.IDimensions
+  ) => void;
+};
+
+type Props<Data, Value> = StaticProps<Data, Value> & State<Data> & Handlers;
+
+export class Cell<Data extends Types.CellBase, Value> extends PureComponent<
   Props<Data, Value>
 > {
   /** @todo update to new API */
   root: HTMLElement | null;
+
+  constructor(props: Props<Data, Value>) {
+    super(props);
+    this.root = null;
+  }
+
   handleRoot = (root: HTMLElement | null) => {
     this.root = root;
   };
 
-  handleMouseDown = (e: SyntheticMouseEvent<HTMLElement>) => {
+  handleMouseDown = (e: React.MouseEvent<HTMLTableCellElement>) => {
```

Comment:
```suggestion
  handleMouseDown = (e: SyntheticMouseEvent<HTMLTableCellElement>) => {
```
Unclear why this was renamed

Reply:
`SyntheticMouseEvent ` Does not exist.